image/internal/manifest/manifest_test.go

```diff
@@ -87,6 +88,9 @@ func TestMatchesDigest(t *testing.T) {
 		{"v2s1.manifest.json", TestDockerV2S2ManifestDigest, false},
 		// Unrecognized algorithm
 		{"v2s2.manifest.json", digest.Digest("md5:2872f31c5c1f62a694fbd20c1e85257c"), false},
+		// SHA512 test cases (these should fail because we're using SHA256 by default)
```

Comment:
No, there’s no reason why `MatchesDigest` shouldn’t be able to validate against `sha512`. It doesn’t, but that can be fixed.